### PR TITLE
refactor cases of "Declarator is reassigned in all

### PR DESCRIPTION
**Pull request by tsteven4**
paths before being read"
when the declarator and the subsequent assignment can be joined.

### CODE REVIEW

File: an1.cc [hunk 1]
```diff
@@ -832,18 +824,13 @@ static void Write_AN1_Waypoints(gbfile* f)
...
-  an1_line_record* rec = nullptr;
-  an1_vertex_record* vert = nullptr;
   route_head* rte_head;
   Waypoint* wpt_tmp;
 
   ReadShort(f);
-  count = ReadLong(f);
-  for (i = 0; i < count; i++) {
-    rec = Alloc_AN1_Line();
+  unsigned long count = ReadLong(f);
```
Comment (GPSBabelDeveloper) [inline an1.cc:831]: Even if count wasn't used, was it needed to seek the read pointer ahead? (On a fone in bed. I may not have the whole picture. I doubt our coverage of an1 is very good.)

**Review by GPSBabelDeveloper — COMMENTED** [review]
Much better. Thanks!

File: an1.cc (hunk 1 above)
Comment (GPSBabelDeveloper) [reply an1.cc:831]: It was the occurrence above in read waypoints that I meant to comment on.
Sorry.

File: an1.cc [hunk 2]
```diff
@@ -684,15 +679,12 @@ static void Write_AN1_Bitmaps(gbfile* f)
...
-  an1_waypoint_record* rec = nullptr;
   Waypoint* wpt_tmp;
   char* icon = nullptr;
   ReadShort(f);
-  count = ReadLong(f);
-  for (i = 0; i < count; i++) {
-    rec = Alloc_AN1_Waypoint();
+  unsigned long count = ReadLong(f);
+  for (unsigned long i = 0; i < count; i++) {
+    an1_waypoint_record* rec = Alloc_AN1_Waypoint();
```
Comment (tsteven4) [inline an1.cc:687]: old line 687-689 declare the variable and initialize it.  The initial value is never used.
old lines 693-695 rewrite all three variables.  The initial values were never read.
new lines 685-687 declare the variables, and use the rewritten values from 693-695, and duplicate the loop.
It looks good to  me.

Comment (robertlipe) [reply an1.cc:687]: Me, too.  I thought I saw a call to ReadLong(f) being dropped, but on a Real Computer, I see it's not. I suspect the diff just rendered badly on the "small" screen .I read it several times. Sorry for the distraction.

**Pull request merged**